reference/matrix/csr_kernels.cpp

```diff
@@ -294,12 +291,8 @@ void advanced_spgemm(std::shared_ptr<const ReferenceExecutor> exec,
     map<IndexType, ValueType> local_row_nzs(exec);
     for (size_type a_row = 0; a_row < num_rows; ++a_row) {
         local_row_nzs.clear();
-        if (vbeta != zero(vbeta)) {
-            spgemm_accumulate_row(local_row_nzs, d, vbeta, a_row);
-        }
-        if (valpha != zero(valpha)) {
-            spgemm_accumulate_row2(local_row_nzs, a, b, valpha, a_row);
-        }
+        spgemm_accumulate_row(local_row_nzs, d, vbeta, a_row);
+        spgemm_accumulate_row2(local_row_nzs, a, b, valpha, a_row);
```

Comment:
no matter what alpha, beta are, always get the csr with the union sparsity of A, B?

Reply:
Yes, I don't really consider the cases `alpha=0` or `beta=0` (effectively scaling all matrix elements) important enough to justify special treatment for them. cuSPARSE/hipSPARSE afaik does the same, unless alpha or beta are `nullptr`.

Reply:
I see.